### DISCUSSION TITLE
[Wayland] Failed to open any DRM device

### DISCUSSION THREAD

**Discussion by raiguard** [qtile/qtile#3043] [Q&A] [answered]
Manjaro updated, and as usually happens when it does, something broke. This time, Qtile decided to start using Wayland! The Xorg option is entirely missing from LightDM, so the only option there is Wayland. When I tried to boot it, it immediately crashed back to LightDM.

I figured out that I need to install the Wayland dependencies, so I installed `wlroots`, `pywlroots`, `pywayland`, and `cairocffi`. I was able to get them installed and to get Qtile to stop complaining about missing dependencies, but now I'm getting an error for which I haven't been able to find any info. Here's a half-decent cell phone pic because I was too frustrated to figure out how to take a screenshot in a TTY:

![image](https://user-images.githubusercontent.com/3515394/142753140-d685f013-ad55-4650-a625-15182bb5a411.png)

A Reddit user who had a somewhat similar issue on Sway said they had to enable something in their kernel, but that's as much detail as they went into.

CPU: AMD Ryzen 3600
GPU: NVIDIA GeForce RTX 2060 Ti
NVIDIA driver version: 495.44-5
Environment: Wayland

I'm probably going to do a fresh vanilla arch install soon, I'm getting sick of Manjaro and the leftover garbage from when I used Gnome for a while.

**Comment by m-col** [comment, marked as answer]
> Manjaro updated, and as usually happens when it does, something broke. This time, Qtile decided to start using Wayland! The Xorg option is entirely missing from LightDM, so the only option there is Wayland. When I tried to boot it, it immediately crashed back to LightDM.

Sounds like they just removed the .desktop file entry - I wonder why...

I'm actually unsure about the error, but as this is happening at the wlroots level it's likely there are some possible solutions over in the sway issue tracker. Here are some that look related: https://github.com/swaywm/sway/issues/4472, https://github.com/swaywm/sway/issues/4672. These suggest that early KMS is required, so check that you have that enabled: https://wiki.archlinux.org/title/Kernel_mode_setting

**Reply by raiguard** [reply]
Alright, enabling early KMS did allow it to boot! However, it flickers like crazy, and the moment I open a program, my mouse cursor disappears and stops working altogether. I tried running with the default config and the same thing is happening.

I know the proprietary Nvidia drivers somewhat support Wayland now, but IIRC they still have issues, so it's possible that that is why this is happening. I'll have to figure out how to re-add the X11 desktop entry so I can still use my PC.

**Reply by raiguard** [reply]
Update: Turns out that manjaro did not delete `qtile.desktop`, it just added a `qtile-wayland.desktop` entry alongside it for some reason. It's no wonder that it wasn't working, because it was in the `xsessions` folder. I removed `qtile-wayland.desktop` and rebooted, and now I can use qtile on X11 again and everything is good!

I'm still probably going to do a clean Arch install though, my system has gotten very messy with tons of duplicate programs as I've been experimenting and learning how to use Linux.
